packages/amazonq/test/unit/amazonq/lsp/lspClient.test.ts

```diff
@@ -0,0 +1,37 @@
+import * as sinon from 'sinon'
+import assert from 'assert'
+import { LspClient } from 'aws-core-vscode/amazonq'
+
+describe('Test LSP client', function () {
+    let lspClient: LspClient
+    let encryptFunc: sinon.SinonSpy
+
+    beforeEach(function () {
+        lspClient = new LspClient()
+        encryptFunc = sinon.spy(lspClient, 'encrypt')
+    })
+
+    it('should encrypt payload of query ', async () => {
+        await lspClient.query('mock_input')
+        assert.ok(encryptFunc.calledOnce)
+        assert.ok(encryptFunc.calledWith(JSON.stringify({ query: 'mock_input' })))
```

Comment:
can the actual encrypted result be tested instead of using a spy?

Reply:
This is done in latest commit

Reply:
Is the spy still needed ?

Reply:
I still need the spy to verify what was encrypted.